pkg/crypto.go

```diff
@@ -70,5 +80,31 @@ func (c Crypto) Setup(hmacFile string, masterFile string) (bool, error) {
 }
 
 func (c Crypto) Protect(value string) (string, error) {
-	return value, nil // TODO implement this
+	var form = map[string]any{"datum": value}
+	log.Infof("%s > Protecting text using crypto.", c.instance.ID())
+	requestFormData := c.instance.http.RequestFormData(form)
+	response, err := requestFormData.Post(CryptoProtectPath)
+
+	if err != nil {
+		return "", fmt.Errorf("%s > cannot encrypt text using crypto: %s", c.instance.ID(), err)
+	} else if response.IsError() {
+		return "", fmt.Errorf("%s > cannot encrypt text using crypto: %s", c.instance.ID(), response.Status())
+	}
+
+	bodyReader := response.RawBody()
+
+	defer func(bodyReader io.ReadCloser) {
```

Comment:
this defer is not needed here.

`ReadCloser` means that when reader/unmarshaller ends reading then the object implementing this interface will also automatically close the source so you don't need to take care;

just pass `response.RawBody()` directly to the `UnmarshalJSON`